docs/docs/misc/glossary/call_types.md

```diff
@@ -0,0 +1,177 @@
+
+#include_code private_call /noir-projects/noir-contracts/contracts/lending_contract/src/main.nr rust
+
+Unlike the EVM however, private execution doesn't revert in the traditional way: in case of error (e.g. a failed assertion, a state changing operation in a static context, etc.) the proof generation simply fails and no transaction request is generated, spending no network gas or user funds.
+
+#### Public Calls
+
+Since public execution can only be performed by the sequencer, public functions cannot be executed in a private context. It is possible however to _enqueue_ a public function call during private execution, requesting the sequencer to run it during inclusion of the transaction.
+
+Since the public call is made asynchronously, any return values or side effects are not available during private execution. If the public function fails once executed, the entire transaction is reverted inncluding state changes caused by the private part, such as new commitments or nullifiers. Note that this does result in gas being spent, like in the case of the EVM.
+
+#include_code enqueue /noir-projects/noir-contracts/contracts/app_subscription_contract/src/main.nr rust
+
+It is also possible to create public functions that can _only_ be invoked by privately enqueing a call from the same contract, which can very useful to update public state after private exection (e.g. update a token's supply after privately minting). This is achieved by annotating functions with `#[aztec(internal)]`.
+
+:::warning
+Calling public functions privately leaks some privacy! The caller of the function and all arguments will be revelead, so exercise care when mixing the private and public domains.
+:::
+
+A common pattern is to enqueue public calls to check some validity condition on public state, e.g. that a deadline has not expired or that some public value is set.
+
+#include_code enqueue_check /noir-projects/noir-contracts/contracts/crowdfunding_contract/src/main.nr rust
+
+To learn about alternative ways to acess public state privately, look into [Shared State](../../developers/contracts/references/storage/shared_state.md).
+
+### Public Execution
+
+Contract functions marked with `#[aztec(public)]` can only be called publicly, and are executed by the sequencer. The computation model is very similar to the EVM: all state, parameters, etc. are known to the entire network, and no data is private.
+
+Since private calls are always run in a user's device, it is not possible to perform any private execution from a public context. A reasonably good mental model for public execution is that of an EVM in which some work has already been done privately, and all that is know about it is its correctness and side-effects (new commitments and nullifiers, enqueued public calls, etc.). A reverted public execution will also revert the private side-effects.
```

Comment:
Instead of new commitments, better saying new notes just to not re-introduce commitment term, think it was tried to keep it more consistent with the note hashes and notes.